include/powerloader/downloader.hpp

```diff
@@ -98,12 +101,11 @@ namespace powerloader
          * @return Return true when another chance to download is allowed.
          */
         bool can_retry_download(int num_of_tried_mirrors, const std::string& url);
-        bool check_msgs(bool failfast);
+        bool check_msgs(bool failfast, bool allow_failure);
```

Comment:
I think it would be clearer to pass an struct with all the options.

Reply:
When we rewrite this part yes. For now I want to keep a red flag when I read this and having many boolean is a good way to do so